setup.cfg

```diff
@@ -0,0 +1,77 @@
+[metadata]
+name = ocrd_webapi_implementation
+version = 0.0.1
+description = Implementation of the OCR-D-webapi
+long_description = file: readme.md
+
+[options]
+packages =
+	ocrd_webapi
+	ocrd_webapi.routers
+install_requires =
```

Comment:
Is this list meant to be exhaustive (including transitive dependencies)? And can you specify version ranges or omit the explicit versions?

Reply:
What I am doing with the versions is just that: I write the code and if I need a library than I install it in the venv via pip. And to write these `install_requires` or requirements.txt I just execute `pip freeze` and put this in requirements.txt or `install_requires`. 
To me it seems your questions assume that there is a deeper sense (or good understanding)  behind what I do here but I have to admit I am mostly just stupidly trying until it works.

Reply:
Haha, no need to be self-critical, `pip freeze` is exactly the right approach in most cases, it guarantees that the set of dependencies work together. With setup.py it was often argued, to generate the `requirements.txt` by doing `pip freeze` and manually edit the `install_requires` in setup.py for less restrictive dependencies. 

So, `pip freeze` is a good start. Then you can get an understanding of the inter-dependencies of packages with `pipdeptree`. Then I would prune all the transitive dependencies, e.g. ocrd brings in Pillow, bagit etc. And finally, unless there is a very strong reason for `==` version pinning, either omit the version or use `>=`.

Nothing pressing, this can be adapted later. I was only curious.